### PR TITLE
[api] Socket Options: do not allow AES GCM if TSBPD is disabled.

### PR DESCRIPTION
**Pull request by maxsharabayko**
The AES-GCM mode cannot be used at the moment if TSBPD is disabled.
The main reason is the different handling of the 'timestamp' field of a data packet (see  #2337).
When TSPBD is disabled, the timestamp is set at the time of sending, unique for every retransmission. The timestamp field is included in the associated data for integrity check, while encryption happens only the first time a packet is sent. Using the AES-GCM would fail on a retransmitted packet.

This PR restricts using AES-GCM without TSBPD until a further decision is taken to either exclude the 'timestamp' field from the integrity check when TSBPD is disabled or (preferably) use the same timestamping rule in any mode.

- Do not allow AES GCM if TSBPD is disabled.
- Do not allow disabling TSBPD if GCM is enabled.

### CODE REVIEW

**Comment by ethouris** [comment]
In the meantime: I think it doesn't hurt to change the data timestamp for the retransmitted packets preserve the original sending time. The reception facility doesn't use timestamps for anything anyway. Of course, this means a change in the protocol, but at least an improvement idea ticket could be started.

File: docs/API/API-socket-options.md [hunk 1]
```diff
@@ -204,7 +204,7 @@ The following table lists SRT API socket options in alphabetical order. Option d
 | [`SRTO_BINDTODEVICE`](#SRTO_BINDTODEVICE)               | 1.4.2 | pre-bind | `string`  |         |                   |          | RW  | GSD+  |
 | [`SRTO_CONGESTION`](#SRTO_CONGESTION)                   | 1.3.0 | pre      | `string`  |         | "live"            | \*       | W   | S     |
 | [`SRTO_CONNTIMEO`](#SRTO_CONNTIMEO)                     | 1.1.2 | pre      | `int32_t` | ms      | 3000              | 0..      | W   | GSD+  |
-| [`SRTO_CRYPTOMODE`](#SRTO_CRYPTOMODE)                   | 1.6.0-dev | pre      | `int32_t` |     | 0 (Auto)          | [0, 3]   | W   | GSD   |
+| [`SRTO_CRYPTOMODE`](#SRTO_CRYPTOMODE)                   | 1.6.0-dev | pre      | `int32_t` |     | 0 (Auto)          | [0, 2]   | W   | GSD   |
```
Comment (ethouris) [inline docs/API/API-socket-options.md:207]: Was that just a typo? Values allowed are 0 and 2 or 0, 1 and 2?

**Comment by maxsharabayko** [comment]
> In the meantime: I think it doesn't hurt to change the data timestamp for the retransmitted packets preserve the original sending time. The reception facility doesn't use timestamps for anything anyway. Of course, this means a change in the protocol, but at least an improvement idea ticket could be started.

Yes, that's the preferable plan for 1.6.0.

File: docs/API/API-socket-options.md (hunk 1 above)
Comment (maxsharabayko) [reply docs/API/API-socket-options.md:207]: 0, 1 and 2. Was a typo.

**Pull request merged**